### ISSUE TITLE
kube-apiserver fails to run if audit-log-path does not end with `audit.log`

### ISSUE THREAD

**Issue by xhejtman** [rancher/rke2#4136]
Hello,

it seems that upgrade from rke2 1.26.3 rke2r1 to 1.26.4 rke2r1 fails because kubeapi cannot start. Using strace I got:
```
{\"message\":\"error reopening /dev/null inside container: open /dev/null: permission denied\"}
```

**Comment by brandond** [comment]
Please fill out the issue template, and include the complete k3s logs.

**Comment by xhejtman** [comment]
It seems that it is caused by `- "audit-log-path=-"` option in rke2 `config.yaml` for kube-api server.

Full kube-api config:
```
kube-apiserver-arg:
        - "encryption-provider-config=/etc/rancher/rke2/encryption-provider-config.json"
        - "audit-log-maxage=30"
        - "audit-log-maxbackup=10"
        - "audit-log-maxsize=100"
        - "audit-log-format=json"
        - "audit-policy-file=/etc/rancher/rke2/audit-policy.yaml"
        - "audit-log-path=-"
```

Logs attached.
[logs.zip](https://github.com/rancher/rke2/files/11302275/logs.zip)

it is 6 node cluster, all nodes are server nodes. It runs on ubuntu 22.04, x86_64.

**Comment by brandond** [comment]
Hmm, indeed. Looks like we'll need to add handling for that special value of the flag.

**Comment by xhejtman** [comment]
Strange, that value works up to 1.26.3.

**Comment by brandond** [comment]
https://github.com/rancher/rke2/pull/4027

I wasn't aware of the special meaning of - for this flag. Usually users want the logs written out to disk somewhere for ingest by a log analysis system.

**Comment by xhejtman** [comment]
Fair enough, we use rancher logging system to collect logs from Pod stdout, but perhaps suggested approach is better.

**Comment by brandond** [comment]
I'll take a look at that, thanks!

**Comment by brandond** [comment]
Oh, there's already special handling in there, but its hardcoded for the default audit log filename :/

Use `--kube-apiserver-arg=audit-log-path=/var/log/kube-audit/audit.log`. Regardless of what directory it's in, the filename must be `audit.log` for it to be handled properly.

**Comment by xhejtman** [comment]
Thanks, it is ok, I can wait for a next release.

**Comment by brandond** [comment]
It should work now if you just call the file audit.log instead of apiserver.log

**Comment by xhejtman** [comment]
Btw, it would be nice, if it gets fixed in 1.24 and 1.25 releases also.
